alfresco-search/doc/architecture/event-trackers/event-based-content-tracker-spike.md

```diff
@@ -0,0 +1,298 @@
+
+![Content Transform Sequence](content-transform-sequence.png)
+
+**Components referenced in the figure**
+
+* [T-Client](https://github.com/Alfresco/alfresco-repository/blob/alfresco-repository-7.60/src/main/java/org/alfresco/messaging/camel/routes/OnContentUpdateRenditionRoute.java#L88): Currently ACS is a transformer client, requesting transformation operations to *Transform Service* via JMS events
+* [Shared File Store](https://github.com/Alfresco/alfresco-shared-file-store): REST API Layer to share contents between a *T-Client* and the *T-Engine*
+* [T-Service](https://github.com/Alfresco/alfresco-transform-service): Alfresco Transform Service to route JMS Events to T-Engines
+* [T-Engine](https://git.alfresco.com/Repository/alfresco-docker-transformers): Transformation Engines provides transformation operations on content, consuming the source from `Shared File Store` and producing the target also in `Shared File Store`. Currently Apache Tika, LibreOffice, ImageMagick and PDF Renderer are available
+
+**Transformation Process**
+
+*T-Client* uploads the source content to `Shared File Store` using a REST API invocation (1) to retrieve a *srcContentRef* (2).
+
+After that, *T-Client* creates a new `TransformRequest` event with the *srcContentRef* and the requested operation (3).
+
+*T-Service* consumes the event and invokes to the right *T-Engine* (LibreOffice, ImageMagick, PdfRenderer, Apache Tika) with the *srcContentRef* and the operation requested (4).
+
+The *T-Engine* performs the transformation (5) and stores the result in the `Shared File Store` to retrieve a *targetContentRef* (6).
+
+*T-Service* creates a new `TransformReply` event with the *targetContentRef* that is consumed by *T-Client* (7), where the transformed content is downloaded from `Shared File Store` using a REST API invocation (8). After that, for instance, this downloaded content is stored in the repository as a new *rendition*, so the temporary copy at `Shared File Store` is not required anymore.
+
+Both source and target files can be deleted from `Shared File Store` after the operation ends successfully (9).
+
+*Note*. Detailed JSON syntax for `Transform Request` and `Transform Reply` are available at [Spec for Transform Request & Reply](https://github.com/Alfresco/alfresco-transform-service/blob/master/docs/transformation-request-reply.md)
+
+
+## Alternatives evaluation
+
+Below different alternatives for integrating Content Tracking with the New Transform Service are described.
+
+
+### 1 - No Change to Search Service
+
+The repository will off load the transformation to Transformation Service and pause the thread requesting the transformation until the content is transformed.
```

Comment:
Does it really pause, i thought it meant to be async?

Reply:
In this first approach, Search Services code is untouched. So this asynchrony is managed by the repository part, which holds the REST API Thread invocation till T-Service has delivered the transformation.

Reply:
I'm not sure this approach is feasible (and already discounted by Repo team, iirc) for an async T-Request/T-Reply. The consumer of the reply will be a different thread (in fact most likely a different Repo node in the cluster).